### PR TITLE
build: update `ng-dev` to fix `caretaker check` and update label config

### PR DESCRIPTION
**Pull request by devversion**
* Updates ng-dev to support the new g3sync canonical JSON configs
* Updates the google-internal-tests option to support the g3 sync canonical JSON config
* Moves the YAML sync config to a dedicated JSON file, used by the action and by `yarn ng-dev`.
* Updates the ng-dev caretaker config to fix that the Merge assistance queue is always empty.

### CODE REVIEW

**Review by josephperrott — APPROVED** [review]
LGTM

Just the one general thought with no action needed.

File: .ng-dev/caretaker.mts [hunk 1]
```diff
@@ -2,14 +2,15 @@ import {CaretakerConfig} from '@angular/ng-dev';
 
 /** The configuration for `ng-dev caretaker` commands. */
 export const caretaker: CaretakerConfig = {
+  g3SyncConfigPath: './.ng-dev/google-sync-config.json',
```
[the remark this answers is not in the document]
Comment (devversion) [reply .ng-dev/caretaker.mts:5]: Agreed, we can standardize this in a follow-up (once we also found a good spot to put this in)

**Comment by Fahstar6** [comment]
MIT

**Comment by Fahstar6** [comment]
خ

**Comment by dylhunn** [comment]
This PR was merged into the repository by commit aff9f4a314b139ba41f203526965c3993d3ff292.

**Pull request closed without merging**